src/runtime_src/core/edge/common_em/config.cxx

```diff
@@ -884,4 +885,46 @@ namespace xclemulation{
     ifs.close();
     return false;
   }
+
+  int getIPName2Index(const char *name,const char* buffer)
+  {
+    std::string errmsg;
+    const uint64_t bad_addr = 0xffffffffffffffff;
+  
+    if (!buffer)
+    {
+      errmsg = "ERROR: getIPName2Index - can't load ip_layout section";
+      std::cout << errmsg << std::endl;
+      return -EINVAL;
+    }
+
+    const ip_layout* map = (reinterpret_cast<const ::ip_layout*>(buffer));
+    if (map->m_count < 0) {
+      errmsg = "ERROR: getIPName2Index - invalid ip_layout section content";
+      std::cout << errmsg << std::endl;
+      return -EINVAL;
+    }
+
+    uint64_t addr = bad_addr;
+    int i;
+    for (i = 0; i < map->m_count; i++) {
+      if (strncmp((char *)map->m_ip_data[i].m_name, name,
+        sizeof(map->m_ip_data[i].m_name)) == 0) {
+        addr = map->m_ip_data[i].m_base_address;
+        break;
+      }
+    }
+    if (i == map->m_count)
+      return -ENOENT;
```

Comment:
Can this case happen without being caught by the next test?
Pick randomly someone and ask this quizz: @sonals can you answer to this question in 10 seconds? :-)

Reply:
agreed, this comparison doesn't make sense. removed this part of code.